### PR TITLE
Document the (CPU|disk|memory) defaults in `multipass help launch` (Fixes #957)

### LINKED ISSUE #957: Document the (CPU|disk|memory) defaults in `multipass help launch`
When writing [my blog post](https://write.wrestle.town/oddbloke/using-multipass-to-create-a-writefreely-dev-vm), I wanted to mention the default amount of memory in a multipass VM and had to launch a new VM to check what it was. It would be nice if I didn't have to do that (and for users on more constrained systems, they may want to know the defaults before they launch at all!).

### PR DESCRIPTION
**Pull request by makspiechota**
Solution for #957

- Moved default values to constants.h, shared between daemon and cli client and used
- Default values printed in launch help
- Unit test to ensure cpu values are convertible to int32 added
- clang-format applied

Note that this is my first pull to your repository (and first open source pull at all) so any comments and suggestions are appreciated

### CODE REVIEW

File: tests/test_constants.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (ricab) [inline tests/test_constants.cpp]: Nice, thanks for the testing

File: tests/test_constants.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#include <gtest/gtest.h>
+
+namespace mp = multipass;
+using namespace testing;
+
+TEST(Constants, constants_constraints) {
+    EXPECT_NO_THROW(std::stoi(mp::min_cpu_cores));
+    EXPECT_NO_THROW(std::stoi(mp::default_cpu_cores));
+}
```
Comment (ricab) [inline tests/test_constants.cpp]: We may as well add a bit more sanity checking. How does this look to you?

```suggestion
    EXPECT_NO_THROW(mp::MemorySize{mp::min_memory_size});
    EXPECT_NO_THROW(mp::MemorySize{mp::default_memory_size});
    EXPECT_NO_THROW(mp::MemorySize{mp::min_disk_size});
    EXPECT_NO_THROW(mp::MemorySize{mp::default_disk_size});
 }

TEST(Constants, defaults_greater_or_equal_to_minimums)
{
    EXPECT_GE(mp::MemorySize{mp::default_memory_size}, mp::MemorySize{mp::min_memory_size});
    EXPECT_GE(mp::MemorySize{mp::default_disk_size}, mp::MemorySize{mp::min_disk_size});
    EXPECT_GE(std::stoi(mp::default_cpu_cores), std::stoi(mp::min_cpu_cores));
}
```

File: tests/test_constants.cpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <multipass/constants.h>
+
```
Comment (ricab) [inline tests/test_constants.cpp]: Include needed for testing proposal
```suggestion
#include <multipass/memory_size.h>

```

File: tests/test_constants.cpp (hunk 2 above)
Comment (makspiechota) [reply tests/test_constants.cpp]: This all make sense, especially now, when I've got familiar with MemorySize constructor.

**Comment by makspiechota** [comment]
Shall I rebase them again on my fork? What about 'co-author' line then, where you are mentioned?

**Comment by Saviq** [comment]
@makspiechota your call, no need to rebase or squash the four commits (it's when there's merges from other branches that it gets unwieldy).

If you do squash, you can still add the "Co-authored-by" line yourself (see [GitHub reference for multiple authors](https://help.github.com/en/articles/creating-a-commit-with-multiple-authors)).

**Comment by makspiechota** [comment]
Alright so I leave it this way and waiting for approval?

**Pull request merged**